### PR TITLE
fix: print-columns and ready condition aggregate

### PR DESCRIPTION
**Pull request by rafalgalaw**
Added some better reasons to conditions and use `Ready` condition as aggregate for all other conditions present. For example if the poolManager is not ready `Ready=False` with a `Reason=ComponentNotReady` and more details provided in the separate `PoolManager` condition:
```yaml
  status:
    conditions:
    - lastTransitionTime: "2024-03-11T13:34:50Z"
      message: Pool Manager is not running
      reason: ComponentNotReady
      status: "False"
      type: Ready
    - lastTransitionTime: "2024-03-11T13:34:50Z"
      message: 'failed to update tools for repo cnt-dev/rthalho-test: fetching tools:
        Unauthorized'
      reason: PoolManagerFailure
      status: "False"
      type: PoolManager
    - lastTransitionTime: "2024-03-05T09:27:24Z"
      message: ""
      reason: FetchingSecretRefSuccess
      status: "True"
      type: SecretReference
    id: 70056770-a244-4fa4-b347-3b5d03cda874

```
and fixed print columns, so with the `-owide` flag one gets more details about the current state
image

### CODE REVIEW

File: pkg/util/conditions/condition_types.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ const (
 	ReconcileErrorReason      ConditionReason = "ReconcileError"
 	DeletingReason            ConditionReason = "Deleting"
 	DeletionFailedReason      ConditionReason = "DeletionFailed"
+	GarmAPIErrorReason        ConditionReason = "GarmAPIError"
+	ComponentNotReadyReason   ConditionReason = "ComponentNotReady"
```
Comment (bavarianbidi) [inline pkg/util/conditions/condition_types.go:18]: `ComponentNotReady` is something i'm currently struggling with. 
All the other errors are self-explained, but `ComponentNotReady` doesn't give me that much information - even with my current knowledge of pools, orgs, etc, ...

Comment (rafalgalaw) [reply pkg/util/conditions/condition_types.go]: fixed with [ecdb745](https://github.com/mercedes-benz/garm-operator/pull/104/commits/ecdb7456b50799937c024916029d0162f01a33d1)

File: pkg/util/conditions/condition_types.go [hunk 2]
```diff
@@ -14,6 +15,9 @@ const (
 	ReconcileErrorReason      ConditionReason = "ReconcileError"
 	DeletingReason            ConditionReason = "Deleting"
 	DeletionFailedReason      ConditionReason = "DeletionFailed"
+	GarmAPIErrorReason        ConditionReason = "GarmAPIError"
+	ComponentNotReadyReason   ConditionReason = "EntityNotReady"
```
Comment (bavarianbidi) [inline pkg/util/conditions/condition_types.go:19]:
```suggestion
```

unused

File: pkg/util/conditions/condition_types.go [hunk 3]
```diff
@@ -38,3 +42,8 @@ const (
 	FetchingSecretRefSuccessReason ConditionReason = "FetchingSecretRefSuccess"
 	FetchingSecretRefFailedReason  ConditionReason = "FetchingSecretRefFailed"
 )
+
+const (
+	GarmServerNotReconciledYet ConditionMessage = "GARM server not reconciled yet"
+	DeletingEnterprise         ConditionMessage = "Deleting enterprise"
+)
```
Comment (bavarianbidi) [inline pkg/util/conditions/condition_types.go:49]: * `DeletingEnterprise` should be used in https://github.com/mercedes-benz/garm-operator/blob/ce290e7ccba2d1e457c2e1ce2e2c8fefdf2c9ebc/internal/controller/enterprise_controller.go#L220
* for `GarmServerNotReconciledYet` see my comments on organization, pool and repository controller

File: internal/controller/repository_controller.go [hunk 4]
```diff
@@ -80,8 +80,9 @@ func (r *RepositoryReconciler) reconcileNormal(ctx context.Context, client garmC
 
 	webhookSecret, err := secret.FetchRef(ctx, r.Client, &repository.Spec.WebhookSecretRef, repository.Namespace)
 	if err != nil {
-		conditions.MarkFalse(repository, conditions.ReadyCondition, conditions.ReconcileErrorReason, err.Error())
+		conditions.MarkFalse(repository, conditions.ReadyCondition, conditions.FetchingSecretRefFailedReason, err.Error())
 		conditions.MarkFalse(repository, conditions.SecretReference, conditions.FetchingSecretRefFailedReason, err.Error())
+		conditions.MarkUnknown(repository, conditions.PoolManager, conditions.UnknownReason, "GARM server not reconciled yet")
```
Comment (bavarianbidi) [inline internal/controller/repository_controller.go:85]:
```suggestion
		conditions.MarkUnknown(repository, conditions.PoolManager, conditions.UnknownReason, conditions.GarmServerNotReconciledYet)
```

File: internal/controller/organization_controller.go [hunk 5]
```diff
@@ -80,8 +80,9 @@ func (r *OrganizationReconciler) reconcileNormal(ctx context.Context, client gar
 
 	webhookSecret, err := secret.FetchRef(ctx, r.Client, &organization.Spec.WebhookSecretRef, organization.Namespace)
 	if err != nil {
-		conditions.MarkFalse(organization, conditions.ReadyCondition, conditions.ReconcileErrorReason, err.Error())
+		conditions.MarkFalse(organization, conditions.ReadyCondition, conditions.FetchingSecretRefFailedReason, err.Error())
 		conditions.MarkFalse(organization, conditions.SecretReference, conditions.FetchingSecretRefFailedReason, err.Error())
+		conditions.MarkUnknown(organization, conditions.PoolManager, conditions.UnknownReason, "GARM server not reconciled yet")
```
Comment (bavarianbidi) [inline internal/controller/organization_controller.go:85]:
```suggestion
		conditions.MarkUnknown(organization, conditions.PoolManager, conditions.UnknownReason, conditions.GarmServerNotReconciledYet)
```

File: internal/controller/enterprise_controller.go [hunk 6]
```diff
@@ -81,8 +81,9 @@ func (r *EnterpriseReconciler) reconcileNormal(ctx context.Context, client garmC
 
 	webhookSecret, err := secret.FetchRef(ctx, r.Client, &enterprise.Spec.WebhookSecretRef, enterprise.Namespace)
 	if err != nil {
-		conditions.MarkFalse(enterprise, conditions.ReadyCondition, conditions.ReconcileErrorReason, err.Error())
+		conditions.MarkFalse(enterprise, conditions.ReadyCondition, conditions.FetchingSecretRefFailedReason, err.Error())
 		conditions.MarkFalse(enterprise, conditions.SecretReference, conditions.FetchingSecretRefFailedReason, err.Error())
+		conditions.MarkUnknown(enterprise, conditions.PoolManager, conditions.UnknownReason, "GARM server not reconciled yet")
```
Comment (bavarianbidi) [inline internal/controller/enterprise_controller.go:86]:
```suggestion
		conditions.MarkUnknown(enterprise, conditions.PoolManager, conditions.UnknownReason, conditions.GarmServerNotReconciledYet)
```

File: pkg/util/conditions/condition_types.go [hunk 7]
```diff
@@ -38,3 +42,8 @@ const (
 	FetchingSecretRefSuccessReason ConditionReason = "FetchingSecretRefSuccess"
 	FetchingSecretRefFailedReason  ConditionReason = "FetchingSecretRefFailed"
 )
+
+const (
+	GarmServerNotReconciledYet ConditionMessage = "GARM server not reconciled yet"
+	DeletingEnterprise         ConditionMessage = "Deleting enterprise"
```
Comment (bavarianbidi) [inline pkg/util/conditions/condition_types.go:48]: If we want to have a `DeletingEnterprise` condition message here, we should also include one for Orgs, Repos and Pools as well

**Review by bavarianbidi — CHANGES_REQUESTED** [review]
few smaller nits - but all in all :+1:

File: internal/controller/runner_controller.go [hunk 8]
```diff
@@ -230,10 +230,6 @@ func (r *RunnerReconciler) EnqueueRunnerInstances(ctx context.Context, instanceC
 		return err
 	}
 
-	if len(pools.Items) < 1 {
-		return nil
-	}
-
```
Comment (bavarianbidi) [inline internal/controller/runner_controller.go:236]: this seems to be here by mistage - as you already propose this change in https://github.com/mercedes-benz/garm-operator/pull/113/files#diff-93df91d9269bd31cc29ac5ba24b171a95bf84f12d51bc82ab4031ee47332b852L233-L236

**Review by bavarianbidi — APPROVED** [review]

**Pull request merged**